File: src/backend/MyAIRunningMate/MyAIRunningMate.Contracts/Aggregates/Responses/AggregateArtifactResponse.cs
using MyAIRunningMate.Contracts.BestEfforts.Responses;

namespace MyAIRunningMate.Contracts.Aggregates.Responses;

public record AggregateArtifactResponse(
    ActivityDetailsResponse ActivityDetails,
    ActivityMetricsResponse? ActivityMetrics,
    IEnumerable<TimeSeriesRecordResponse>? TimeSeriesRecords,
    IEnumerable<LapViewResponse> Laps,
    IEnumerable<BestEffortResponse>? BestEfforts
);

File: src/frontend/MyAIRunningMate-UI/src/types/aggregates/aggregateArtifactResponse.ts
import type { BestEffortResponse } from "../dashboard/bestEffortResponse";
import type { ActivityDetailsResponse } from "./activityDetailsResponse";
import type { ActivityMetricsResponse } from "./activityMetricsResponse";
import type { LapResponse } from "./lapResponse";
import type { TimeSeriesRecordResponse } from "./timeSeriesRecordResponse";


export interface AggregateArtifactResponse {
  activity_details: ActivityDetailsResponse;
  activity_metrics: ActivityMetricsResponse;
  time_series_records: TimeSeriesRecordResponse[];
  laps: LapResponse[];
  best_efforts: BestEffortResponse[] | null;
}